dev/core#5566 Fix clear options not showing for non-admin user

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
dev/core#5566 Fix clear options not showing for non-admin user

Before
----------------------------------------
When a non admin user views a profile with a radio custom field in it with options_per_line set to 2 the clear link is missing. Depending on how the profile is accessed it may also be missing for non-admin users

![image](https://github.com/user-attachments/assets/fd60856a-27e8-4bc5-b31f-5cdc7f8fb6b8)


After
----------------------------------------
![image](https://github.com/user-attachments/assets/83c55c86-fb58-4d97-8cde-a237e9066217)

Technical Details
----------------------------------------
This is confusing cos replication is hard - in part due to caching & also because viewing profiles in different ways accesses different tpls

1) the Dynamic tpl - this is at (e.g) https://dmaster.localhost:32353/civicrm/profile/create?gid=1&reset=1
2) the fields.tpl - this is accessed at (e.g) https://dmaster.localhost:32353/civicrm/admin/uf/group/preview?reset=1&gid=1&fieldId=1

Prior to this change the edit & clear never rendered at all in the former tpl and in the latter tpl the clear was not displayed unless the edit was ALSO displayed. This changes it so they both share a tpl snippet and the clear should always be present

Comments
----------------------------------------

eileenmcnaughton committed:
Rename in-template variable for clarity

'n' is not very meaningful. The 'sister' template uses profileFieldName - so let's align

eileenmcnaughton committed:
Fix clear radio not showing for non-admin user

eileenmcnaughton commented on templates/CRM/UF/Form/FieldOptionsPerLineRadio.tpl:
```diff
@@ -0,0 +1,20 @@
+    {* Include the edit options list for admins *}
+    {if $formElement.html|strstr:"crm-option-edit-link"}
+      {$formElement.html|regex_replace:"@^.*(<a href=.*? class=.crm-option-edit-link.*?</a>)$@":"$1"}
+    {else}
```
This else is the actual change - all the rest is just adjusting the 2 templates to share the same code

seamuslee001 commented on templates/CRM/UF/Form/FieldRadioOptionsPerLine.tpl:
```diff
@@ -0,0 +1,20 @@
+    {if $formElement.html|strstr:"crm-option-edit-link"}
+      {$formElement.html|regex_replace:"@^.*(<a href=.*? class=.crm-option-edit-link.*?</a>)$@":"$1"}
+    {else}
+      {$formElement.html}
```
This seems like it might duplicate things? I don't see this in the fields.tpl nor in dynamic.tpl previously

seamuslee001 replied:
or is this just the wrench?

eileenmcnaughton replied:
@seamuslee001 yeah so what was happening is that because the radio options were not being laid out as desired the tpl was iterating through & rendering the options html not the radio group html - and then using that to re-add some of the radio group html (but was dropping the clear when the edit was not in play)

Although this PR does mostly get us there I wound up concluding there is a better approach - ie update the renderer to add the classes - see https://github.com/civicrm/civicrm-core/pull/31706 (as of writing the PR name is not a good description of what it does)

eileenmcnaughton committed:
Do not re-render the html in the error for radio fields with option values

This does the same as https://github.com/civicrm/civicrm-packages/pull/416
but in Civi code rather than in packages.

My take on it is that the original sin is that the quickform class for rendering
radio options does not nest them in a div / allow us to add classes to that div
and hence we are working around it by rendering the radio options
individually in the template layer.

If we fix that maybe we revert this - it is mostly a case of
registering a class with more suitable markup in
HTML_QuickForm::registerElementType()
